core/domain/wipeout_service.py

```diff
@@ -112,70 +130,110 @@ def pre_delete_user(user_id):
+            user_id, False, False, False, False)
 
-    email = user_services.get_user_settings(user_id, strict=True).email
     user_services.mark_user_for_deletion(user_id)
-
-    save_pending_deletion_request(
+    pending_deletion_requests.append(
         wipeout_domain.PendingDeletionRequest.create_default(
             user_id,
-            email,
+            user_settings.email,
+            user_settings.role,
             explorations_to_be_deleted_ids,
             collections_to_be_deleted_ids
         )
     )
 
+    save_pending_deletion_requests(pending_deletion_requests)
+
 
 def delete_user(pending_deletion_request):
-    """Delete all the models for user specified in pending_deletion_request.
+    """Delete all the models for user specified in pending_deletion_request
+    on the basis of the user role specified in the request.
 
     Args:
         pending_deletion_request: PendingDeletionRequest. The pending deletion
             request object for which to delete or pseudonymize all the models.
     """
-    _delete_user_models(pending_deletion_request.user_id)
-    _hard_delete_explorations_and_collections(pending_deletion_request)
-    _delete_improvements_models(pending_deletion_request.user_id)
-    _delete_activity_models(
-        pending_deletion_request,
-        models.NAMES.question,
-        question_models.QuestionSnapshotMetadataModel,
-        question_models.QuestionCommitLogEntryModel,
-        'question_id')
-    _delete_activity_models(
-        pending_deletion_request,
-        models.NAMES.skill,
-        skill_models.SkillSnapshotMetadataModel,
-        skill_models.SkillCommitLogEntryModel,
-        'skill_id')
-    _delete_activity_models(
-        pending_deletion_request,
-        models.NAMES.story,
-        story_models.StorySnapshotMetadataModel,
-        story_models.StoryCommitLogEntryModel,
-        'story_id')
+    if pending_deletion_request.role == feconf.ROLE_ID_LEARNER:
+        _delete_profile_user_models(pending_deletion_request.user_id)
+    else:
+        _delete_full_user_models(pending_deletion_request.user_id)
+        _hard_delete_explorations_and_collections(pending_deletion_request)
+        _delete_improvements_models(pending_deletion_request.user_id)
+        _delete_activity_models(
+            pending_deletion_request,
+            models.NAMES.question,
+            question_models.QuestionSnapshotMetadataModel,
+            question_models.QuestionCommitLogEntryModel,
+            'question_id')
+        _delete_activity_models(
+            pending_deletion_request,
+            models.NAMES.skill,
+            skill_models.SkillSnapshotMetadataModel,
+            skill_models.SkillCommitLogEntryModel,
+            'skill_id')
+        _delete_activity_models(
+            pending_deletion_request,
+            models.NAMES.story,
+            story_models.StorySnapshotMetadataModel,
+            story_models.StoryCommitLogEntryModel,
+            'story_id')
```

Comment:
```suggestion
        _delete_full_user(pending_deletion_request)
```
and move all this code to that method.

Reply:
Done.